src/main/java/com/coveros/selenified/element/check/azzert/AssertMatches.java

```diff
@@ -0,0 +1,173 @@
+
+    public AssertMatches(Element element, OutputFile file) {
+        this.element = element;
+        this.file = file;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public OutputFile getOutputFile() {
+        return file;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Element getElement() {
+        return element;
+    }
+
+    // ///////////////////////////////////////
+    // assessing functionality
+    // ///////////////////////////////////////
+
+    /**
+     * Asserts that the element's text matches the regular expression pattern provided. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedPattern the expected pattern of the text of the element
+     */
+    public void text(String expectedPattern) {
+        String text = checkText(expectedPattern, 0, 0);
+        assertNotNull(NO_ELEMENT_FOUND, text);
+        assertTrue("Text Mismatch: text of '" + text + DOES_NOT_MATCH_PATTERN + expectedPattern + "'", text.matches(expectedPattern));
+    }
+
+    /**
+     * Asserts that the element's pattern in a particular cell matches the regular expression
+     * pattern provided. If the element isn't present, or a table, this will
+     * constitute a failure, same as a mismatch. This information will be logged
+     * and recorded, with a screenshot for traceability and added debugging
+     * support.
+     *
+     * @param row     - the number of the row in the table - note, row numbering
+     *                starts at 1, NOT 0
+     * @param col     - the number of the column in the table - note, column
+     *                numbering starts at 1, NOT 0
+     * @param pattern - what pattern do we expect to be in the table cell
+     */
+    public void text(int row, int col, String pattern) {
+        String text = checkText(row, col, pattern, 0, 0);
+        assertNotNull(NO_ELEMENT_FOUND, text);
+        assertTrue("Text Mismatch: text of '" + text + DOES_NOT_MATCH_PATTERN + pattern + "'", text.matches(pattern));
+    }
+
+    /**
+     * Asserts that the element's value matches the regular expression pattern
+     * provided. If the element isn't present or an input, this will constitute a failure,
+     * same as a mismatch. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param expectedPattern the expected input value of the element
+     */
+    public void value(String expectedPattern) {
+        String value = checkValue(expectedPattern, 0, 0);
+        if (value == null) {
+            String reason = NO_ELEMENT_FOUND;
+            if (getElement().is().present()) {
+                reason = "Element not input";
+            }
+            fail(reason);
+        }
+        assertNotNull(value);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "value" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS2259)